petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -237,3 +239,57 @@ def test_change_cache_dir_raise_error(test_ctx):
     # restore conf (other test need use it)
     test_ctx.spark.conf.set('petastorm.spark.converter.parentCacheDirUrl', test_ctx.temp_url)
     assert test_ctx.temp_url == _get_parent_cache_dir_url()
+
+
+def test_tf_dataset_batch_size(test_ctx):
+    df1 = test_ctx.spark.range(100)
+
+    batch_size = 30
+    converter1 = make_spark_converter(df1)
+
+    with converter1.make_tf_dataset(batch_size=batch_size) as dataset:
+        iterator = dataset.make_one_shot_iterator()
+        tensor = iterator.get_next()
+        with tf.Session() as sess:
+            ts = sess.run(tensor)
+    assert len(ts.id) == batch_size
+
+
+@contextmanager
+def mock_make_batch_reader():
+    captured_args = []
+    import petastorm.spark
+
+    def mock_fn(dataset_url, **kwargs):
+        reader_args = {'dataset_url': dataset_url}
+        reader_args.update(kwargs)
+        captured_args.append(reader_args)
+        return make_batch_reader(dataset_url, **kwargs)
+
+    petastorm.spark.spark_dataset_converter.make_batch_reader = mock_fn
+    try:
+        yield captured_args
+    finally:
+        petastorm.spark.spark_dataset_converter.make_batch_reader = make_batch_reader
+
+
+def test_tf_dataset_petastorm_args(test_ctx):
```

Comment:
Is there a reason not to use  `mock.patch` here? there won't be need in a custom `mock_make_batch_reader` implementation.